[Vinoth Kumar Kunasilan] iP

noth-k opened the pull request:
### Buddy&I

> "You got a friend in me" - Woody

Buddy helps you manage your tasks more efficiently for you :)

- text based
- Intuitive

Features

- [x] Adding, deleting tasks
- [x] Marking, unmarking tasks

Follow these steps for easy usage:
1. Open it up
2. Add tasks
3. Live stress free about remembering these items and [study](https://www.youtube.com/shorts/vQDLOxfOpSc)

**FREE (special offer)**😍

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

noth-k committed:
Level-0

noth-k committed:
Level-1

noth-k committed:
Level-2

noth-k committed:
Level-3

noth-k committed:
Level-4

noth-k committed:
Level-5

noth-k committed:
Level-6

noth-k committed:
A-Enums

noth-k committed:
A-TextUiTesting

noth-k committed:
Level-7

noth-k committed:
Merge commit branch-Level-7

noth-k committed:
Level-8

noth-k committed:
Merge branch 'branch-Level-8'

noth-k committed:
A-MoreOOP

noth-k committed:
Merge pull request #1 from noth-k/branch-A-MoreOOP

A-MoreOOP

noth-k committed:
A-Packages

noth-k committed:
Merge branch 'add-gradle-support'

noth-k committed:
A-JUnit

noth-k committed:
A-Jar

noth-k committed:
A-JavaDoc

noth-k committed:
Level-9

noth-k committed:
merged JavaDoc, Codingstandard, Level-9

noth-k committed:
merged JavaDoc, Codingstandard, Level-9

noth-k committed:
merged JavaDoc, Codingstandard, Level-9 v2

jeremychiaaaaa commented on src/main/java/Buddy.java:
```diff
@@ -0,0 +1,42 @@
+import command.Command;
+import exceptions.BuddyException;
+import parser.Parser;
+import storage.Storage;
+import task.TaskList;
+import ui.Ui;
```
Should there should be a line spacing between the different groups of imports?

jeremychiaaaaa commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,65 @@
+package parser;
+
+import command.AddCommand;
+import command.FindCommand;
+import command.ExitCommand;
+import command.DeleteCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.UnmarkCommand;
+import command.Command;
```
Should these be in lexographical order?

jeremychiaaaaa commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,65 @@
+        String mainCommand = commandParts[0].toLowerCase();
+
+        switch (mainCommand) {
+            case "todo":
+            case "deadline":
+            case "event":
+                return new AddCommand(command);
+            case "delete":
+                return new DeleteCommand(parseTaskIndex(commandParts));
+            case "mark":
+                return new MarkCommand(parseTaskIndex(commandParts));
+            case "unmark":
+                return new UnmarkCommand(parseTaskIndex(commandParts));
+            case "bye":
+                return new ExitCommand();
+            case "list":
+                return new ListCommand();
+            case "find":
+                return new FindCommand(commandParts[1].trim());
+            default:
```
Should the case and default lines be in-line with the Switch statement as according to the coding standards?

jeremychiaaaaa commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,157 @@
+import task.Events;
+import task.Task;
+import task.ToDos;
+
```
I like how you grouped the import statements

jeremychiaaaaa commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,157 @@
+     * @throws BuddyException If the file is not found or if there is an error parsing the file contents.
+     */
+    public ArrayList<Task> load() throws BuddyException {
+        ArrayList<Task> list = new ArrayList<>();
```
Should 'list' be in plural form?

jeremychiaaaaa reviewed (commented):
Overall, great job on following the coding standards! I highlighted some small points for your consideration.

ryenl commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    public void displayWelcome() {
+        System.out.println("    ---------------------------------------------------------");
```
Would it be useful to implement a showLine method in the Ui class for this?

ryenl commented on src/test/java/TodoStub.java:
```diff
@@ -0,0 +1,31 @@
+import task.Task;
+public class TodoStub extends Task {
```
Maybe add a line spacing here?

ryenl commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,122 @@
+        System.out.println("    Bye! See ya soon!");
+    }
+
+    public void displayUpperLineBreak() {
```
Good naming of Ui methods to easily distinguish functionality!

ryenl commented on src/test/java/TodoTest.java:
```diff
@@ -0,0 +1,14 @@
+import org.junit.jupiter.api.Test;
+import task.ToDos;
+
+
```
line spacing

ryenl reviewed (commented):
Just very minor nitpicks. Overall very well written!

noth-k committed:
Add functionality for Level-10: Implement GUI

noth-k committed:
Add assertions

noth-k committed:
Fix code quality by applying DRY principle

noth-k committed:
Merge pull request #2 from noth-k/branch-A-Assertions

Add assertions

noth-k committed:
Merge branch 'master' into branch-A-CodeQuality

noth-k committed:
Merge pull request #3 from noth-k/branch-A-CodeQuality

Fix code quality by applying DRY principle

noth-k committed:
Add update functionality

noth-k committed:
Merge pull request #4 from noth-k/branch-C-Update

Add update functionality

noth-k committed:
Add error handling

noth-k committed:
Add user guide

noth-k committed:
Update user guide

User Guide is not displaying Ui demonstration. Ui.png is added to enhance user guide

noth-k committed:
Add window title

noth-k committed:
Update user guide

noth-k committed:
Add Ui image

noth-k committed:
Update README.md
